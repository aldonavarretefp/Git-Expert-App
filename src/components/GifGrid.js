import React from 'react'
import PropTypes from 'prop-types'


import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

import 'animate.css'

const GifGrid = ({category}) => {
    
    const {data:images,loading} = useFetchGifs(category);

    return (
        <>
            <h2 className='animate__animated animate__zoomInUp'>{category}</h2>
            {
                loading && <h2 className='animate__animated animate__flash'>Cargando...</h2>
            }
            <div className="card-grid">
                {
                    images.map(({id,title,url}) => (
                        <GifGridItem key={id} id={id} title={title} url={url}/>
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
