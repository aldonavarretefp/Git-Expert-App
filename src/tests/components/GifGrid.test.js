import React from 'react';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';

describe('Pruebas en <GifGrid />', () => {
    let wrapper = shallow(<GifGrid />);
    test('debe mostrar <GifGrid/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
});