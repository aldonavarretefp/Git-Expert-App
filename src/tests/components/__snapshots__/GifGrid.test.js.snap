// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar <GifGrid/> correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__zoomInUp"
  />
  <h2
    className="animate__animated animate__flash"
  >
    Cargando...
  </h2>
  <div
    className="card-grid"
  />
</Fragment>
`;
